agent/handlers/v3/container_metadata_handler.go

```diff
@@ -37,10 +39,66 @@ func ContainerMetadataHandler(state dockerstate.TaskEngineState) func(http.Respo
 			utils.WriteJSONToResponse(w, http.StatusBadRequest, responseJSON, utils.RequestTypeContainerMetadata)
 			return
 		}
-
+		containerResponse, err := GetContainerResponse(containerID, state)
+		if err != nil {
+			errResponseJSON, _ := json.Marshal(err.Error())
+			utils.WriteJSONToResponse(w, http.StatusBadRequest, errResponseJSON, utils.RequestTypeContainerMetadata)
+			return
+		}
 		seelog.Infof("V3 container metadata handler: writing response for container '%s'", containerID)
 
-		// v3 handler shares the same container metadata response format with v2 handler.
-		v2.WriteContainerMetadataResponse(w, containerID, state)
+		responseJSON, _ := json.Marshal(containerResponse)
+		utils.WriteJSONToResponse(w, http.StatusOK, responseJSON, utils.RequestTypeContainerMetadata)
+	}
+}
+
+// GetContainerResponse gets container response for v3 metadata
+func GetContainerResponse(containerID string, state dockerstate.TaskEngineState) (*v2.ContainerResponse, error) {
+	containerResponse, err := v2.NewContainerResponse(containerID, state)
```

Comment:
the network information is populated in  v2's `response.go` only if the task has an ENI, since it should work only for awsvpc tasks. So I added logic here to add network response for non awsvpc tasks